src/transformers/modeling_utils.py

```diff
@@ -1534,8 +1534,16 @@ def resize_token_embeddings(
             return model_embeds
 
         # Update base model and current model config
-        self.config.vocab_size = model_embeds.weight.shape[0]
-        self.vocab_size = model_embeds.weight.shape[0]
+        # get the actual size (considering both zero stage 3 and pad_to_multiple_of)
+        if is_deepspeed_zero3_enabled():
+            import deepspeed
+
+            with deepspeed.zero.GatheredParameters(model_embeds.weight, modifier_rank=None):
+                new_num_tokens = model_embeds.weight.shape[0]
+        else:
+            new_num_tokens = model_embeds.weight.shape[0]
```

Comment:
This part is actually already handled in the underlying method `_resize_token_embeddings`. I propose we assign the new value to `self.config.vocab_size` and `self.vocab_size` directly within `_resize_token_embeddings`, or that we have `_resize_token_embeddings` updated so that it returns the final `new_num_tokens` value

Reply:
@LysandreJik sure. I just wanted to have the minimal fix and avoid larger changes to the program logic or signatures, but it seems to be safe enough. Feel free to open another PR on top and refactor as you like.